### ISSUE TITLE
Add finding from template: KeyError: 'req_resp'

### ISSUE THREAD

**Issue by valentijnscholten** [DefectDojo/django-DefectDojo#2886]
While looking at 2818, I noticed something else that goes wrong before that error:

- Create 'a finding template by going to "Finding Templates" --> Add a Finding Template'
- Go To 'Product --> engagement --> Test and "add finding from Template" '
- The Application 'Provides with a list of all the finding templates, --> select Use this Template'

```
Internal Server Error: /test/20478/add_findings/2
Traceback (most recent call last):
  File "/home/valentijn/venv_dd/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/valentijn/venv_dd/lib/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/valentijn/venv_dd/lib/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/valentijn/venv_dd/lib/python3.6/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/c/Data/django-DefectDojo/dojo/test/views.py", line 579, in add_temp_finding
    'tags': [tag.name for tag in finding.tags]})
  File "/c/Data/django-DefectDojo/dojo/forms.py", line 1007, in __init__
    req_resp = kwargs.pop('req_resp')
KeyError: 'req_resp'
Internal Server Error: /test/20478/add_findings/2
Traceback (most recent call last):
  File "/home/valentijn/venv_dd/lib/python3.6/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/home/valentijn/venv_dd/lib/python3.6/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/valentijn/venv_dd/lib/python3.6/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/valentijn/venv_dd/lib/python3.6/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/c/Data/django-DefectDojo/dojo/test/views.py", line 579, in add_temp_finding
    'tags': [tag.name for tag in finding.tags]})
  File "/c/Data/django-DefectDojo/dojo/forms.py", line 1007, in __init__
    req_resp = kwargs.pop('req_resp')
KeyError: 'req_resp'
```

**Comment by Maffooch** [comment]
Goodness that form is used everywhere. This is something that also should have been caught by unit tests.. Very strange. Will look into it.

**Comment by valentijnscholten** [comment]
I think there is no integration test to add a finding from a template. Only applying a template to a finding. Would be good to have that test.

**Comment by Maffooch** [comment]
So I looked into this and found that there is big bug around the risk acceptance stuff. When checking for risk acceptances, there is no test attached to the finding yet, so the whole thing stops there. Don't want to break your stuff here.

All that's needed for the request response stuff is to pass a keyword arg as follows to the FindingForm in the add_temp_finding function `req_resp=None`. This is because templates cannot hold those fields without a model change, and they are also pretty unique to fit into a template anyways.

**Comment by valentijnscholten** [comment]
The risk thing is 2818 and fixed in #2861. I don't think it's a bug in the risk stuff. The apply template is providing an empty instance to the form which I don't think is how that should work. It worked fine for add finding which uses the normal way.

**Comment by valentijnscholten** [comment]
Fixed it by adding some if statement. Still think the empty finding that is used in the form is wrong. technical debt from the past :-)
